### PR TITLE
Modifications for web batch

### PR DESCRIPTION
**Pull request by BettyB979**

### CODE REVIEW

File: src/main/java/fr/insee/queen/batch/dao/mongo/impl/DataDaoMongoImpl.java [hunk 1]
```diff
@@ -58,16 +59,18 @@ public JSONObject getDataBySurveyUnitId(String suId) throws ParseException {
...
 		Query query = new Query();
 		query.addCriteria(Criteria.where("id").in(lstSu));
 		List<SurveyUnit> suList = mongoTemplate.find(query, SurveyUnit.class, "survey_unit");
+		AtomicInteger nb= new AtomicInteger();
 		suList.stream().forEach(su -> {
 			if(su != null && su.getData() != null) {
 				mongoTemplate.remove(su.getData(), Constants.DATA);
+				nb.getAndIncrement();
 			}
 		});
```
Comment (SimonDmz) [inline src/main/java/fr/insee/queen/batch/dao/mongo/impl/DataDaoMongoImpl.java:72]: same Atomic ☢️ pattern 😄

File: src/main/java/fr/insee/queen/batch/dao/mongo/impl/ParadataEventDaoMongoImpl.java [hunk 2]
```diff
@@ -16,69 +14,84 @@
...
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
 
 /**
  * Service for the Paradata-Event entity that implements the interface associated
- * @author scorcaud
  *
+ * @author scorcaud
```
Comment (SimonDmz) [inline src/main/java/fr/insee/queen/batch/dao/mongo/impl/ParadataEventDaoMongoImpl.java:25]: denounce yourself here plz 👮‍♂️

File: src/main/java/fr/insee/queen/batch/dao/mongo/impl/ParadataEventDaoMongoImpl.java [hunk 3]
```diff
@@ -16,69 +14,84 @@
...
-		public ParadataEvent(UUID id, JSONObject value) {
-			this.id = id;
-			this.value = value;
-		}
-	}
+        public ParadataEvent(UUID id, JSONObject value) {
+            this.id = id;
+            this.value = value;
+        }
+    }
```
Comment (SimonDmz) [inline src/main/java/fr/insee/queen/batch/dao/mongo/impl/ParadataEventDaoMongoImpl.java:96]: api/queen/batch/object/ParadataEvent => duplicate ?

File: src/main/java/fr/insee/queen/batch/dao/impl/jpa/ParadataEventDaoJpaImpl.java [hunk 4]
```diff
@@ -45,24 +47,45 @@ public class ParadataEventDaoJpaImpl implements ParadataEventDao{
...
-		StringBuilder qString = new StringBuilder("SELECT value FROM paradata_event WHERE EXISTS ( select 1 ")
-				.append("from jsonb_each_text(value) as t(")
-				.append(databaseService.getKeyParadataIdSu())
-				.append(",v) where v ilike ?)");
-		value =  jdbcTemplate.queryForList(qString.toString(), new Object[]{suId}, String.class);
-		for(String val : value) {
-			JSONObject jsonObjectTemp = (JSONObject) parser.parse(val);
+		JSONArray arrayIds = new JSONArray();
+		StringBuilder qString = new StringBuilder("SELECT * FROM paradata_event WHERE value->>'" +
+				this.databaseService.getKeyParadataIdSu()+"' =  ? ");
```
[the remark this answers is not in the document]
Comment (BettyB979) [reply src/main/java/fr/insee/queen/batch/dao/impl/jpa/ParadataEventDaoJpaImpl.java:52]: fixed

**Pull request merged**